src/objectExplorer/objectExplorerService.ts

```diff
@@ -65,21 +65,22 @@ export class ObjectExplorerService {
                 let nodeConnection = this._sessionIdToConnectionCredentialsMap.get(result.sessionId);
                 for (let connection of savedConnections) {
                     if (Utils.isSameConnection(connection.connectionCreds, nodeConnection)) {
-                        nodeLabel = connection.label;
-                        break;
+                        // if it's not the defaul label
+                        if (connection.label !== connection.connectionCreds.server) {
```

Comment:
What is this fixing?

Also don't you want to break out if the connection is the same regardless of whether the label is the default one or not?

Reply:
This is fixing the fact that root nodes can now be databases as well, not just servers. So, we only change the label if there's an actual profile name, not just the server name.